src/connection/request.rs

```diff
@@ -1,9 +1,106 @@
 use crate::Response;
+use crate::sql::worker::SQL;
+use crate::sql::worker::SQLError;
+use crate::sql::parser::Parser;
+use crate::sql::parser::ParserError;
+use std::fmt;
+
+
+#[derive(Debug)]
+pub struct Request {
+}
+
+
+#[derive(Debug)]
+pub enum RequestMsg {
```

Comment:
Error 和 Succeed 的設計要分開

```rust
    let responses = messages.map(move |message| match Request::parse(&message, &mut sql) {
        Ok(req) => req, // Here is successful response 
        Err(e) => return Response::Error { msg: e }, // Here is error response
}
```

Or here we can just leave error type as empty, since we need to give feedback anyway, so all responces are `Ok()`.

`Result<Response, ()>` might be OK.

`Result<Response, ()>`  or `Result<Response, ResponseError>` depends on your choice.
But never put String in Error field. Error should format in string later when it happens. Before that, it should be translated as Error chain.

Reply:
I mean this line 
```rust
pub fn parse(input: &str, mut sql: &mut SQL) -> Result<Response, String> 
```